src/module.c

```diff
@@ -5678,12 +5737,34 @@ RedisModuleCallReply *RM_Call(RedisModuleCtx *ctx, const char *cmdname, const ch
 
         if (ctx->client->user == NULL) {
             errno = ENOTSUP;
+            if (error_as_call_replies) {
+                sds msg = sdsnew("ERR acl verification failed, context is not attached to a client.");
+                reply = callReplyCreateError(msg, ctx);
+            }
             goto cleanup;
         }
         acl_retval = ACLCheckAllUserCommandPerm(ctx->client->user,c->cmd,c->argv,c->argc,&acl_errpos);
         if (acl_retval != ACL_OK) {
             sds object = (acl_retval == ACL_DENIED_CMD) ? sdsdup(c->cmd->fullname) : sdsdup(c->argv[acl_errpos]->ptr);
             addACLLogEntry(ctx->client, acl_retval, ACL_LOG_CTX_MODULE, -1, ctx->client->user->name, object);
+            if (error_as_call_replies) {
+                sds msg = NULL;
+                switch (acl_retval) {
+                case ACL_DENIED_CMD:
+                    msg = sdsnew("ERR acl verification failed, the user can't run this command or subcommand.");
+                    break;
+                case ACL_DENIED_KEY:
+                    msg = sdsnew("ERR acl verification failed, the user can't access at least one of the keys mentioned in the command arguments.");
+                    break;
+                case ACL_DENIED_CHANNEL:
+                    msg = sdsnew("ERR acl verification failed, the user can't publish to the channel mentioned in the command.");
+                    break;
+                default:
+                    msg = sdsnew("ERR acl verification failed, the user lacking the permissions for the command.");
+                    break;
```

Comment:
maybe that's our opportunity to unify all these switch cases in some way (already exists in scripts, multi, processCommand, and ACL DRYRUN)

Reply:
Fixed